src/internal/util/argsArgArrayOrObject.ts

```diff
@@ -10,21 +10,29 @@ const { getPrototypeOf, prototype: objectProto, keys: getKeys } = Object;
 export function argsArgArrayOrObject<T, O extends Record<string, T>>(args: T[] | [O] | [T[]]): { args: T[]; keys: string[] | null } {
   if (args.length === 1) {
     const first = args[0];
-    if (isArray(first)) {
-      return { args: first, keys: null };
-    }
-    if (isPOJO(first)) {
-      const keys = getKeys(first);
-      return {
-        args: keys.map((key) => first[key]),
-        keys,
-      };
+    const result = arrayOrObject(first);
+    if (result) {
+      return result;
     }
   }
 
   return { args: args as T[], keys: null };
 }
 
+export function arrayOrObject<T, O extends Record<string, T>>(first: T | T[] | O) {
```

Comment:
@benlesh- do you want to split this out (per @demensky)?

Reply:
Nah. It's fine in the same file for now. Every file we add actually adds a little overhead when bundling.